app/test/features/calendar_sync/calendar_sync_test.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:acter/features/calendar_sync/calendar_sync.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:device_calendar/device_calendar.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../../helpers/mock_a3sdk.dart';
+
+import 'package:mocktail/mocktail.dart';
+
+class MockDeviceCalendarPlugin extends Mock implements DeviceCalendarPlugin {}
+
+void main() {
+  group('Calendar Sync test', () {
+    late DeviceCalendarPlugin deviceCalendarFallback;
+    setUpAll(() {
+      deviceCalendarFallback = deviceCalendar;
+      deviceCalendar = MockDeviceCalendarPlugin();
+    });
+
+    tearDownAll(() {
+      deviceCalendar = deviceCalendarFallback;
+    });
+
+    testWidgets('ensure refresh synchronicity', (tester) async {
+      SharedPreferences.setMockInitialValues({}); // no values set yet.
+
+      const calendarId = '1';
+
+      final events = generateMockCalendarEvents(10);
+      final firstTry =
+          events.map((e) => (event: e, rsvp: RsvpStatusTag.Maybe)).toList();
+      final secondTry =
+          events.map((e) => (event: e, rsvp: RsvpStatusTag.No)).toList();
+      final thirdTry =
+          events.map((e) => (event: e, rsvp: RsvpStatusTag.Yes)).toList();
+
+      int updateCountId = 0;
+      // we start fresh
+      when(() => deviceCalendar.createOrUpdateEvent(any()))
+          .thenAnswer((a) async {
+        final r = Result<String>();
+        r.data = 'resultKey-$updateCountId';
+        updateCountId += 1;
+        return r;
+      });
+
+      // create in fast succession;
+      final waiter = Future.wait([
+        scheduleRefresh(calendarId, firstTry),
+        scheduleRefresh(calendarId, secondTry),
+        scheduleRefresh(calendarId, thirdTry),
+      ]);
+
+      // advance the timer
+      await tester.pump(const Duration(seconds: 2));
```

Comment:
this is what happened before: we were called multiple times in a very short time frame with slightly different input, mainly on RSVP status and because it was executed simultaneously, the mapping as well as the actual device events got out of sync.


this test does that, then advances below the debounce threshold and confirms nothing happened, then advances way after that threshold and ensures it was only executed _once_ and has the right mapping saved.